.github/workflows/verify.yml

```diff
@@ -0,0 +1,66 @@
+name: Verify
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  test:
+    runs-on: ubuntu-16.04
+    timeout-minutes: 40
+
+    strategy:
+      fail-fast: true
+      matrix:
+        ruby:
+          - 2.5
+          - 2.6
+          - 2.7
+        test_cmd:
+          - bundle exec rake spec
+          - bundle exec rake cucumber
+          - bundle exec rake yard
+
+    env:
+      RAILS_ENV: test
+
+    name: Ruby ${{ matrix.ruby }} - ${{ matrix.test_cmd }}
+    steps:
+      - name: Install system dependencies
+        run: sudo apt-get install graphviz
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+
+      - name: Setup bundler
+        run: |
+          gem install bundler
+
+      - uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gems-
+
+      - name: Bundle install
+        run: |
+          bundle config path vendor/bundle
+          bundle install --jobs 4 --retry 3
+        env:
+          BUNDLER_WITHOUT: coverage development pcap
+
+      - name: ${{ matrix.test_cmd }}
+        run: |
+          echo "${CMD}"
+          bash -c "${CMD}"
+        env:
+          CMD: ${{ matrix.test_cmd }}
```

Comment:
Let's update this in the same way as metasploit-credentials

---

Ran this gem's tests locally and it generates the coverage artifacts under the coverage folder:
![image](https://user-images.githubusercontent.com/60357436/116388528-d2d5bd00-a813-11eb-8cba-807019b9b311.png)

I think we should be able to add this step and have it available for download:
```yaml
      - name: Upload coverage report
        uses: actions/upload-artifact@v2
        with:
          name: coverage-${{ matrix.ruby }}
          path: |
            coverage/
          retention-days: 1
```

So we can extract all of the coverage reports directly from github
![image](https://user-images.githubusercontent.com/60357436/116390186-b33f9400-a815-11eb-8bec-ae16658452e3.png)

https://github.com/rapid7/metasploit-credential/pull/152/files#r622038016